File: src/Introduction/HomeWorks/HW3/Task3.java
package Introduction.HomeWorks.HW3;

import java.util.ArrayList;

public class Task3 {

    public static void solution() {
        System.out.println("3. Задан целочисленный список ArrayList. Найти минимальное, максимальное и среднее из этого списка.");

        ArrayList<Integer> numbers = GetList.getList(10);
        Print.printList("Input list", numbers);
        Integer minValue = getMinValue(numbers);
        Print.printInteger("Min value: " , minValue);
        Integer maxValue = getMaxValue(numbers);
        Print.printInteger("Max value: " , maxValue);
        Integer averageValue = getAverageValue(numbers);
        Print.printInteger("Average value: " , averageValue);
    }

    private static Integer getAverageValue(ArrayList<Integer> list) {
        Integer averageValue = 0;
        for (Integer number : list) {
            averageValue += number;
        }
        return averageValue / list.size();
    }

    private static Integer getMaxValue(ArrayList<Integer> list) {
        Integer maxValue = Integer.MIN_VALUE;
        for (Integer number : list) {
            if (maxValue < number) maxValue = number;
        }
        return maxValue;
    }

    private static Integer getMinValue(ArrayList<Integer> list) {
        Integer minValue = Integer.MAX_VALUE;
        for (Integer number : list) {
            if (minValue > number) minValue = number;
        }
        return minValue;
    }


}

File: src/Introduction/HomeWorks/HW3/Kotlin/Task3.kt
package Introduction.HomeWorks.HW3.Kotlin

class Task3 {
    fun solution(): Unit {
        println("\n3. Задан целочисленный список ArrayList. Найти минимальное, максимальное и среднее из этого списка.\n")

        val numbers: ArrayList<Int> = GetNumbers.get(10)
        println("Input list:\n$numbers\n")

        val minValue: Int = getMinValue(numbers)
        println("Min value: $minValue")

        val maxValue: Int = getMaxValue(numbers)
        println("Max value: $maxValue")

        val averageValue: Int = getAverageValue(numbers)
        println("Average value: $averageValue")
    }

    private fun getAverageValue(list: java.util.ArrayList<Int>): Int {
        var averageValue: Int = 0;
        for (number in list) averageValue += number

        return averageValue / list.size
    }

    private fun getMaxValue(list: java.util.ArrayList<Int>): Int {
        var maxValue: Int = Int.MIN_VALUE
        for (number in list) {
            if (maxValue < number) maxValue = number
        }
        return maxValue
    }

    private fun getMinValue(list: java.util.ArrayList<Int>): Int {
        var minValue: Int = Int.MAX_VALUE
        for (number in list) {
            if (minValue > number) minValue = number
        }
        return minValue
    }
}